src/js/sharedEth/connectors.js

```diff
@@ -149,23 +145,24 @@ async function connectShowAccount(eth, showAccount) {
   };
 }
 
-async function connectWalletConnect(eth, disallowAuthDialog = false, desiredNetworkId = 1) {
-  const ethProviderName = desiredNetworkId == 3 ? 'ropsten' : 'mainnet';
-  const JSONRPC_URL = eth.dataProviders[ethProviderName].host;
-  const CHAIN_ID = desiredNetworkId;
-
-  const trxProvider = new WalletConnectProvider({
-    rpc: { [CHAIN_ID]: JSONRPC_URL },
-  });
+async function connectWalletConnect(eth, disallowAuthDialog = false, desiredNetworkId = 1, walletConnectProjectId) {
+  let trxProvider;
 
   try {
+    trxProvider = await EthereumProvider.init({
```

Comment:
From the doc here https://docs.walletconnect.com/2.0/web/providers/ethereum#initialization
there are 5 required fields `methods ` and `events` are required as well? 

Also do we need `rpcMap ` for other chains since we are multi-chain now

Reply:
https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#ethereum-provider

These docs for the upgrade seem to the be the only ones to get it working. WC's team confirmed also looking at the code. Their docs sucks.